src/systemctl/systemctl.c

```diff
@@ -2428,12 +2428,25 @@ static int unit_file_find_path(LookupPaths *lp, const char *unit_name, char **un
 
         STRV_FOREACH(p, lp->search_path) {
                 _cleanup_free_ char *path;
+                _cleanup_free_ char *lpath = NULL;
+                int r;
 
                 path = path_join(arg_root, *p, unit_name);
                 if (!path)
                         return log_oom();
 
-                if (access(path, F_OK) == 0) {
+                if (access(path, F_OK) != 0)
+                        continue;
+
+                r = readlink_and_make_absolute(path, &lpath);
```

Comment:
hmm, please use something based on chase_symlinks() here, so that the root path is properly taken into account! We finally have that functionality in systemd, so that we can follow symlinks correctly relative to some root directory, hence we really should use it for cases like this.